server/container_create.go

```diff
@@ -292,23 +303,33 @@ func setupContainerUser(specgen *generate.Generator, rootfs, mountLabel, ctrRunD
 		return err
 	}
 
-	// verify uid exists in containers /etc/passwd, else generate a passwd with the user entry
-	passwdPath, err := utils.GeneratePasswd(uid, gid, rootfs, ctrRunDir)
-	if err != nil {
-		return err
+	genPasswd := true
+	for _, mount := range specgen.Config.Mounts {
+		if mount.Destination == "/etc" ||
+			mount.Destination == "/etc/" ||
+			mount.Destination == "/etc/passwd" {
+			genPasswd = false
```

Comment:
We can add a break here when we find one of these to avoid looking at the rest.

Reply:
Fixed